.clang-format

```diff
@@ -0,0 +1,138 @@
+---
+Language:        Cpp
+# BasedOnStyle:  Mozilla
+AccessModifierOffset: -3
+AlignAfterOpenBracket: Align
+AlignArrayOfStructures: Left
+AlignConsecutiveMacros: false
+AlignConsecutiveAssignments: true
+AlignConsecutiveDeclarations: true
+AlignEscapedNewlines: Right
+AlignOperands:   true
+AlignTrailingComments: true
+AllowAllArgumentsOnNextLine: true
+AllowAllConstructorInitializersOnNextLine: true
+AllowAllParametersOfDeclarationOnNextLine: false
+AllowShortBlocksOnASingleLine: Empty
+AllowShortCaseLabelsOnASingleLine: true
+AllowShortFunctionsOnASingleLine: Inline
+AllowShortLambdasOnASingleLine: All
+AllowShortIfStatementsOnASingleLine: Never
+AllowShortLoopsOnASingleLine: false
+AlwaysBreakAfterDefinitionReturnType: None
+AlwaysBreakAfterReturnType: None
+AlwaysBreakBeforeMultilineStrings: false
+AlwaysBreakTemplateDeclarations: Yes
+BinPackArguments: false
+BinPackParameters: false
+BraceWrapping:
+  AfterCaseLabel:  false
+  AfterClass:      false
+  AfterControlStatement: Never
+  AfterEnum:       false
+  AfterFunction:   false
+  AfterNamespace:  false
+  AfterObjCDeclaration: false
+  AfterStruct:     false
+  AfterUnion:      false
+  AfterExternBlock: false
+  BeforeCatch:     false
+  BeforeElse:      false
+  IndentBraces:    false
+  SplitEmptyFunction: false
+  SplitEmptyRecord: false
+  SplitEmptyNamespace: true
+BreakBeforeBinaryOperators: None
+BreakBeforeBraces: Custom
+BreakBeforeInheritanceComma: false
+BreakInheritanceList: BeforeComma
+BreakBeforeTernaryOperators: true
+BreakConstructorInitializersBeforeComma: false
+BreakConstructorInitializers: BeforeComma
+BreakAfterJavaFieldAnnotations: false
+BreakStringLiterals: true
+ColumnLimit:     120
```

Comment:
ColumnLimit of 120 is needlessly low. For example, my 14" laptop comfortably displays 140+ columns while my desktop display show 252. 

Please select a value somewhere in the range of 140 to 180.  Personally, I like 140 but it may be too small for others.

Reply:
Scott, the issue with long lines is that many developers like to use multiple editors side to side. I use up to three.
With very long lines occasionally, most of the screen is occupied by empty space. Some developers like this set to `80`, as is the case at Ripple (see https://github.com/greg7mdp/rippled/blob/72f6e44ed711a6ebe106810f0b08b59a5695d5dc/.clang-format#L37), but I think this causes excessive statement folding.
I think `120` is a reasonable compromise, but we can see if a consensus emerges.

Reply:
Note: I use two 4k monitors and in emacs each shows 413 columns of text. But again I like using multiple emacs franmes side-by-side.

Reply:
This is a duplicate of an unresolved peer-review comment above, but I will add this...why should every Leap developer limit their lines to 120 columns to accommodate _your_ development setup?

Reply:
@kj4ezj the reason I proposed this setting is obviously *not* to accommodate my development setup. I kind of resent you implying that. This is what I truly feel is the best compromise for all C++ developers (and btw I think any number between ~100 and ~130 would be equally fine). Other languages may have different sweet spots. I am certainly not trying to force my own preferences on other developers.

I was just trying to explain that the reason I believe a line width limit is beneficial is *not* that I am working on a small screen, but that it benefits everybody, regardless of the screen size.

Many companies limit the line length to 80 characters, including [Google](https://google.github.io/styleguide/cppguide.html#Line_Length). I've used that setting at Ripple, and I do find it is very constraining.

On the other hand, with unlimited line length, there will be occasionally very long line which, depending on your editor setting, will either:
- fold out the line, breaking the visual code indentation
- not display the end of the line, preventing you to see all the code at once

or of course you can set a very wide editor window just for the sake of the occasional long line, but then most of your screen is wasted space with no information.

This is why I proposed 120 characters. I feel it is a good compromise, and doesn't create the same hassle as a 80 character limit does when trying to fit C++ statements on one line.

Reply:
Lets consolidate this discussion to a single thread. Since this was marked `resolved` by the author, I will reply in the other thread.